sphinx/_cli/__init__.py

```diff
@@ -0,0 +1,298 @@
+            '\n',
+        ]
+        return ''.join(help_fragments)
+
+    @staticmethod
+    def _format_option_string(option_strings: Sequence[str]) -> str:
+        # hide 'colour'
+        option_strings = [o for o in option_strings if o != '--colour']
+        prefix = '    ' * all(o[1] == '-' for o in option_strings)
+        return prefix + '  ' + ', '.join(map(bold, option_strings))
+
+    @staticmethod
+    def _format_metavar(
+        nargs: int | str | None,
+        metavar: str | tuple[str, ...] | None,
+        choices: Iterable[str] | None | None,
+        dest: str,
+    ) -> str:
+        if metavar is not None:
+            metavar = metavar
+        elif choices is not None:
+            metavar = '{' + ', '.join(sorted(choices)) + '}'
+        else:
+            metavar = dest.upper()
+        if nargs is None:
+            return f'{metavar}'
+        elif nargs == argparse.OPTIONAL:
+            return f'[{metavar}]'
+        elif nargs == argparse.ZERO_OR_MORE:
+            if len(metavar) == 2:
+                return f'[{metavar[0]} [{metavar[1]} ...]]'
+            else:
+                return f'[{metavar} ...]'
+        elif nargs == argparse.ONE_OR_MORE:
+            return f'{metavar} [{metavar} ...]'
+        elif nargs == argparse.REMAINDER:
+            return '...'
+        elif nargs == argparse.PARSER:
+            return f'{metavar} ...'
+        raise ValueError("invalid nargs value")
+
+    def error(self, message: str) -> NoReturn:
+        sys.stderr.write(__(
+            "{0}: error: {1}\n"
+            "Run '{0} --help' for information"  # NoQA: COM812
+        ).format(self.prog, message))
+        raise SystemExit(2)
+
+
+def _create_parser() -> _RootArgumentParser:
+    parser = _RootArgumentParser(
+        prog='sphinx',
+        description=__('   Manage documentation with Sphinx.'),
+        epilog=__('For more information, visit https://www.sphinx-doc.org/en/master/man/.'),
+        add_help=False,
+        allow_abbrev=False,
+    )
+    parser.add_argument(
+        '-V', '--version',
+        action='store_true',
+        default=argparse.SUPPRESS,
+        help=__('Show the version and exit.'),
+    )
+    parser.add_argument(
+        '-h', '-?', '--help',
+        action='store_true',
+        default=argparse.SUPPRESS,
+        help=__('Show this message and exit.'),
+    )
+    parser.add_argument(
+        '-v', '--verbose',
+        action='count',
+        dest='verbosity',
+        default=0,
+        help=__('Increase verbosity (can be repeated)'),
+    )
+    parser.add_argument(
+        '-q', '--quiet',
+        action='store_const',
+        dest='verbosity',
+        const=-1,
+        help=__('Only print errors and warnings.'),
+    )
+    parser.add_argument(
+        '--silent',
+        action='store_const',
+        dest='verbosity',
+        const=-2,
+        help=__('No output at all'),
+    )
+    parser.add_argument(
+        '--colour', '--color',
```

Comment:
A true patriot would choose either :gb: or :us: (or better :ireland: :wink:).

Kidding aside, I'd just choose one and deal with it. Also, does this need to be tertiary? What would you expect `--color yes` to do if the terminal didn't support it? Perhaps `--color/--no-color` would be better, with the implication that the former only does something if it's supported? Alternatively, if you think we should allow requesting a colour and fail if we can't enable it, perhaps the `auto` behavior could be the default and not explicitly configurable via CLI flags (i.e. you'd still just have `--color/--no-color` and it would default to the former only if supported)

Reply:
I've changed my mind on this and I slightly prefer an approach similar to Ruff's ([link](https://beta.ruff.rs/docs/faq/#how-can-i-disable-ruffs-color-output)) -- don't provide a CLI flag, but use environment variables. This makes the behaviour always 'auto', and you can force-disable colour by setting ``NO_COLOR=1``.

If you think we should provide a colour CLI flag I'm happy to implement as you suggest, though.